include/z64camera.h

```diff
@@ -835,56 +841,72 @@ typedef struct {
+    /*  5 */ CAM_ITEM_TYPE_5, // drop bugs and blue fire from bottle
+    /*  8 */ CAM_ITEM_TYPE_8=8, // ??? get item after underwater?
+    /*  9 */ CAM_ITEM_TYPE_9, // get item
+    /* 10 */ CAM_ITEM_TYPE_10, // used farore's wind or nayru's love
+    /* 11 */ CAM_ITEM_TYPE_11, // talking to navi?
+    /* 12 */ CAM_ITEM_TYPE_ONEPOINTDEMO9806,
+    /* 81 */ CAM_ITEM_TYPE_HORSE=81,
+    /* 90 */ CAM_ITEM_TYPE_90=90, // play ocarina (on its own)?
+    /* 91 */ CAM_ITEM_TYPE_91 // play ocarina for an actor?
+} CameraItemType;
+
+typedef struct {
+    /* 0x00 */ f32 atToEyeTargetStepYaw;
+    /* 0x04 */ f32 atToEyeTargetStepPitch;
+    /* 0x08 */ f32 unk_08; // unused
+    /* 0x0C */ s16 atToEyeTargetYaw;
+    /* 0x0E */ s16 atToEyeTargetPitch;
+    /* 0x10 */ s16 animTimer;
+    /* 0x12 */ s16 unk_12; // unused
+    /* 0x14 */ s16 itemType;
 } KeepOn4ReadWriteData; // size = 0x18
 
 typedef struct {
     /* 0x00 */ KeepOn4ReadOnlyData roData;
     /* 0x20 */ KeepOn4ReadWriteData rwData;
 } KeepOn4; // size = 0x38
 
-#define KEEPON4_FLAG_0 (1 << 0)
-#define KEEPON4_FLAG_1 (1 << 1)
-#define KEEPON4_FLAG_2 (1 << 2)
-#define KEEPON4_FLAG_3 (1 << 3)
-#define KEEPON4_FLAG_4 (1 << 4)
-#define KEEPON4_FLAG_5 (1 << 5)
-#define KEEPON4_FLAG_6 (1 << 6)
-#define KEEPON4_FLAG_7 (1 << 7)
+#define KEEPON4_FLAG_NO_CHECK_COL (1 << 0) // If set, disables checking for colliders and collision to place the camera eye. Never set
+#define KEEPON4_FLAG_EYE_YAW_REL_TO_PLAYER (1 << 1) // pitch: provided, yaw: offset from behind player
+#define KEEPON4_FLAG_EYE_ABS (1 << 2) // pitch: provided, yaw: provided. Unused
+#define KEEPON4_FLAG_EYE_FROM_TARGET (1 << 3) // pitch and yaw: depends on x/y rotations of the camera `target`, plus offsets
+#define KEEPON4_FLAG_ONEPOINTDEMO9806 (1 << 4) // Camera timer ticks down, and other things (TODO). Only set for CAM_ITEM_TYPE_ONEPOINTDEMO9806
+#define KEEPON4_FLAG_NOOP (1 << 5) // No effect. Only set for CAM_ITEM_TYPE_11
+#define KEEPON4_FLAG_EYE_KEEP_YAW (1 << 6) // pitch: provided, yaw: retain current yaw
+#define KEEPON4_FLAG_HORSE (1 << 7) // (partly a KEEPON4_FLAG_EYE_) pitch: provided, yaw: offset from the side opposite to the camera `target` compared to player
 
-#define CAM_FUNCDATA_KEEP4(yOffset, eyeDist, pitchTarget, yawTarget, atOffsetZ, fov, interfaceField, yawUpdateRateTarget, unk_22) \
+#define CAM_FUNCDATA_KEEP4(yOffset, eyeDist, pitchTarget, yawTarget, atOffsetPlayerForwards, fov, interfaceField, yawUpdateRateTarget, initTimer) \
     { yOffset, CAM_DATA_Y_OFFSET }, \
     { eyeDist, CAM_DATA_EYE_DIST }, \
     { pitchTarget, CAM_DATA_PITCH_TARGET }, \
     { yawTarget, CAM_DATA_YAW_TARGET }, \
-    { atOffsetZ, CAM_DATA_AT_OFFSET_Z }, \
+    { atOffsetPlayerForwards, CAM_DATA_AT_OFFSET_Z }, \
     { fov, CAM_DATA_FOV }, \
     { interfaceField, CAM_DATA_INTERFACE_FIELD }, \
     { yawUpdateRateTarget, CAM_DATA_YAW_UPDATE_RATE_TARGET }, \
-    { unk_22, CAM_DATA_UNK_22 }
+    { initTimer, CAM_DATA_UNK_22 }
```

Comment:
Or perhaps `UNK_22` was suppose to be the init timer, and the one above is wrong?

Reply:
bump